client-js/client/client.ts

```diff
@@ -353,15 +354,44 @@ export class PipecatClient extends RTVIEventEmitter {
     await this._transport.initDevices();
   }
 
+  /**
+   * startBot() is a method that initiates the bot by posting to a specified endpoint
+   * that optionally returns connection parameters for establishing a transport session.
+   * @param startBotParams
+   * @returns Promise that resolves to TransportConnectionParams or unknown
+   */
+  public async startBot(
+    startBotParams: APIEndpoint
+  ): Promise<TransportConnectionParams | unknown> {
+    if (
+      ["authenticating", "connecting", "connected", "ready"].includes(
+        this._transport.state
+      )
+    ) {
+      throw new RTVIErrors.RTVIError(
+        "Voice client has already been started. Please call disconnect() before starting again."
+      );
+    }
+    this._transport.state = "authenticating";
+    this._abortController = new AbortController();
+    const response = await getTransportConnectionParams(
```

Comment:
now that we're being more opinionated/prescriptive about the fact that the endpoint that gets the connection params _is_ the "start bot" endpoint, updating this method name accordingly would read more clearly, imo

Reply:
something along the lines of `callStartBotEndpoint` (not wedded to that name, we just what first popped into my head)

Reply:
I was actually thinking of going the other way with this and renaming it to something like `makeRequest`

Reply:
updating that as well to be generic and moving the `StartBotError` creation into `startBot()` itself...